ddclient

```diff
@@ -4256,7 +4254,7 @@ sub nic_nsupdate_update {
         my $zone       = $config{$h}{'zone'};
         my $ip         = $config{$h}{'wantip'};
         my $recordtype = '';
-        if (is_ipv6($ip)) {
+        if (ipv6_match($ip)) {
```

Comment:
Why make this change? `is_ipv6` looks more correct to me.

Reply:
So thats the debate I'd like to have.  Is it worthwhile having `is_ipv4` and `is_ipv6` as separate functions when using the `ipvN_match` functions can accomplish much the same thing?  It is certainly easier reading to come across a test for `is_something` but then should we have these functions just for readability?

The issue that prompts me to ask for the debate is that I updated the `ipvN_match` functions to strip out embedded leading zeros.  Which means that if you test `is_ipv4("192.168.001.001")` then it is going to return false.  As long as we are comfortable with this then I would keep the functions and use them -- especially as 99% of the time what we're testing came from `get_ip` -- inside of which we should remove any embedded leading zeros.

Reply:
I decided to leave `is_ipv4` and `is_ipv6` in and use them (so I changed this back to how it was).